nodejs/week1/homework/src/backend/fetchData.js

```diff
@@ -0,0 +1,28 @@
+const reviews = require('./data/reviews');
+const meals = require('./data/meals'); //{ ...obj, reviews: [] }
+
+//  add reviews to all meals and empty to what meal without reviews
+const reviewsInMeals = meals.map(meal => ({
+  ...meal,
+  reviews: findReviewsForMeals(meal.id),
+}));
+
+function findReviewsForMeals(mealID) {
+  return reviews.filter(review => review.mealId === mealID);
+}
+// meals with written reviews 
+const fullReviewedMeal = 
```

Comment:
I'm not sure what you are doing here. For the `/meal` route, you can just grab a random meal from your `reviewsInMeals` array :)

Reply:
Thanks Markus, some of the meals inside reviewsInMeals  has black reviews,, and actually this piece of code add reviews to the reviewsInMeals array and if there is no reviews for that ID it add black : ' ' to the array,,, and with random meal we need that meal to have a review!